src/runAll.js

```diff
@@ -0,0 +1,80 @@
+/* eslint no-console: 0 */
+
+'use strict'
+
+const sgf = require('staged-git-files')
+const Listr = require('listr')
+const stringifyObject = require('stringify-object')
+const getConfig = require('./getConfig')
+const runScript = require('./runScript')
+const generateTasks = require('./generateTasks')
+const resolveGitDir = require('./resolveGitDir')
+
+module.exports = function runAll(packageJson, originalConfig) {
+  const config = getConfig(originalConfig)
+  const gitDir = config.gitDir
+  const verbose = config.verbose
+  const concurrent = config.concurrent
+  const renderer = config.renderer
+  sgf.cwd = resolveGitDir(gitDir)
+
+  if (verbose) {
+    console.log(`
+Running lint-staged with the following config:
+${stringifyObject(config)}
+
+`)
+  }
+
+  return new Promise((resolve, reject) => {
```

Comment:
The promise returned from here is not being used. Results in this:

```
(node:4740) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): undefined
(node:4740) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Also, maybe use `Promise.resolve().then()`? Manual invocation of `resolve`/`reject` wouldn't be required.